libraries/iio/iio.c

```diff
@@ -742,6 +696,103 @@ ssize_t iio_step(struct iio_desc *desc)
 	return tinyiiod_read_command(desc->iiod);
 }
 
+/* Get string for channel id from channel type */
+static char *get_channel_id(enum iio_chan_type type)
+{
+	switch (type) {
+	case IIO_VOLTAGE:
+		return "voltage";
+	case IIO_CURRENT:
+		return "current";
+	case IIO_ALTVOLTAGE:
+		return "altvoltage";
+	}
+
+	return "";
+}
+
+/*
+ * Generate an xml describing a device and write it to buff.
+ * Will return the size of the xml.
+ * If buff_size is 0, no data will be written to buff, but size will be returned
+ */
+static uint32_t iio_generate_device_xml(struct iio_device *device, char *name,
+					int32_t id, char *buff,
+					uint32_t buff_size)
+{
+	struct iio_channel	*ch;
+	struct iio_attribute	*attr;
+	char			ch_id[50];
+	int32_t			i;
+	int32_t			j;
+	int32_t			k;
+	int32_t			n;
+
+	if (buff_size == -1)
+		n = 0;
+	else
+		n = buff_size;
+
+	if (buff == NULL)
+		/* Set dummy value for buff. It is used only for counting */
+		buff = ch_id;
+
+	i = 0;
+	i += snprintf(buff, max(n - i, 0),
```

Comment:
snprintf may return a negative number and you assume all returns are positive

Reply:
I researched a bit and from the documentation, negative error will be returned when a **encoding ** error occurs and encoding only occurs when using unicode strings with "%ls". I use only ascii string so no encoding happens.
I tested with all ascii characters and no error.